backend/infrahub/graphql/auth/query_permission_checker/super_admin_checker.py

```diff
@@ -0,0 +1,34 @@
+from infrahub.auth import AccountSession
+from infrahub.core import registry
+from infrahub.core.branch import Branch
+from infrahub.core.constants import GlobalPermissions
+from infrahub.database import InfrahubDatabase
+from infrahub.graphql import GraphqlParams
+from infrahub.graphql.analyzer import InfrahubGraphQLQueryAnalyzer
+
+from .interface import CheckerResolution, GraphQLQueryPermissionCheckerInterface
+
+
+class SuperAdminPermissionChecker(GraphQLQueryPermissionCheckerInterface):
+    permission_required = f"global:{GlobalPermissions.SUPER_ADMIN.value}:allow"
+
+    async def supports(self, db: InfrahubDatabase, account_session: AccountSession, branch: Branch) -> bool:
+        return account_session.authenticated
+
+    async def check(
+        self,
+        db: InfrahubDatabase,
+        account_session: AccountSession,
+        analyzed_query: InfrahubGraphQLQueryAnalyzer,
+        query_parameters: GraphqlParams,
+        branch: Branch,
+    ) -> CheckerResolution:
+        is_super_admin = False
+        for permission_backend in registry.permission_backends:
+            is_super_admin = await permission_backend.has_permission(
+                db=db, account_id=account_session.account_id, permission=self.permission_required, branch=branch
+            )
+            if is_super_admin:
+                break
```

Comment:
I think it might be a bit cleaner to return here instead of breaking.

I.e. `return CheckerResolution.TERMINATE` then the last return can be just `return CheckerResolution.NEXT_CHECKER` making it a bit easier to read.

Reply:
Good catch